### PR TITLE
automatically seed reference guides from config

### PR DESCRIPTION
**Pull request by tim-dot-org**
Adds the call to seed_all to get each environment in sync once config files are on levelbuilder.

## Links

- jira ticket: [PLAT-1699](https://codedotorg.atlassian.net/browse/PLAT-1699)

### CODE REVIEW

**Comment by dmcavoy** [comment]
I hadn't taken in that we were going to serialize and seed these in a different way that vocab and resources (the other two things associated with course_version). Thinking through this I think what you have here is better because it:

- Does not add more stuff to the script_json files which are already huge
- allows us to seed reference guides after course versions have been set up so we don't have the same needed to seed twice issue we have for vocab and resources

My concerns are:

- If we decided to directly associate levels or lessons with reference guides then they would need to be seeded before levels, scripts, and unit_groups which would not be as simple as reordering the order since the reference guide holds the association to the course_version and not the other way around

I think @bethanyaconnor and @davidsbailey should weigh in here as well.

**Comment by bethanyaconnor** [comment]
This is a departure from how other parts of the curriculum model work, I think this becomes a question of whether to break away from storing everything in script_json or try to keep everything together. My very high level opinion is that the top level model should seed children models, so in that world, vocab and resources would move out of script_json (I always found it weird that they were only serialized if they were in a lesson or were a teacher resource).

Re the circular dependency: if we are planning on adding these to lessons, or adding a more direct association with levels, we end up trading one circular dependency (course version not being present when reference guides are being seeded) for another (reference guides not being available when levels/lessons are being seeded). If those aren't in the immediate plans, it's definitely a vote to seeding these on their own to avoid that problem. If not, then maybe it's worth keeping the seeding logic consistent.

File: dashboard/lib/tasks/seed.rake [hunk 1]
```diff
@@ -449,7 +453,7 @@ namespace :seed do
     sh('mysqldump -u root -B dashboard_test > db/ui_test_data.sql')
   end
 
-  FULL_SEED_TASKS = [:check_migrations, :videos, :concepts, :scripts, :courses, :callouts, :school_districts, :schools, :secret_words, :secret_pictures, :ap_school_codes, :ap_cs_offerings, :ib_school_codes, :ib_cs_offerings, :state_cs_offerings, :donors, :donor_schools, :foorms].freeze
+  FULL_SEED_TASKS = [:check_migrations, :videos, :concepts, :scripts, :courses, :reference_guides, :callouts, :school_districts, :schools, :secret_words, :secret_pictures, :ap_school_codes, :ap_cs_offerings, :ib_school_codes, :ib_cs_offerings, :state_cs_offerings, :donors, :donor_schools, :foorms].freeze
   UI_TEST_SEED_TASKS = [:check_migrations, :videos, :concepts, :course_offerings_ui_tests, :scripts_ui_tests, :courses_ui_tests, :callouts, :school_districts, :schools, :secret_words, :secret_pictures, :donors, :donor_schools].freeze
```
Comment (bethanyaconnor) [inline dashboard/lib/tasks/seed.rake:457]: Are there any plans to add reference guide UI tests here?

Comment (tim-dot-org) [reply dashboard/lib/tasks/seed.rake:457]: no immediate plans, but that would probably be a good idea. I'll make a ticket and note to make sure data gets seeded for that

**Comment by davidsbailey** [comment]
thanks for tagging me @dmcavoy . overall I like the implementation that Tim has here where ref guides are seeded separately from scripts. putting them after `courses` makes sense to me because (IIRC) that's when we create course versions.

> My concerns are:
> * If we decided to directly associate levels or lessons with reference guides then they would need to be seeded before levels, scripts, and unit_groups which would not be as simple as reordering the order since the reference guide holds the association to the course_version and not the other way around

my thoughts on this are:
1. I agree that we couldn't avoid circular dependencies right now if we were to start supporting lesson <--> ref guide connections today
2. we should plan for a future where we seed course versions explicitly, right after course offerings. the work to start doing this is one of the "do no matter what" outcomes from the CourseVersion + UnitGroup work that Bethany wrote up. once this is done, the new seed order could be: course offering, course version, ref guide, scripts/lessons.
3. to go a step further, we could simply consider the course version seeding work to be a prereq for adding lesson <--> ref guide associations in the future. 
4. the solution for level <--> ref guide relationships seems similar, except that we might have to move course offering + course version seeding to before level seeding

TL;DR: explicitly seeding course version should eliminate circular dependency problems for ref guide seeding.

> This is a departure from how other parts of the curriculum model work, I think this becomes a question of whether to break away from storing everything in script_json or try to keep everything together. My very high level opinion is that the top level model should seed children models, so in that world, vocab and resources would move out of script_json (I always found it weird that they were only serialized if they were in a lesson or were a teacher resource).

I agree that it would make more logical sense and perhaps be simpler to work with if we stored resource and vocab definitions outside of script_json. my assumption is that we'd still keep lesson-resource and lesson-vocab relationships within the .script_json. I see two reasons for moving in this direction:
1. minimize code complexity
2. minimize script json size

because vocab and resources are small in json, I don't feel super strongly about prioritizing moving them out of script json until / unless we find that it is causing an issue for us in terms of complexity. perhaps this is something we can keep in mind as we proceed with the course version refactoring work, as well as the work to allow units to be moved or copied in and out of unit groups (I forget exactly what we have planned there). if anyone feels strongly about doing this sooner I'm open to pushback -- perhaps we can prioritize it against the rest of our non-blocking tech debt.

> the top level model should seed children models

@bethanyaconnor can you help me understand what you mean by this -- are you just referring to seeding course version explicitly, before scripts?

**Comment by bethanyaconnor** [comment]
> are you just referring to seeding course version explicitly, before scripts

Yep, that's the gist. I'm also kind of saying, depending on a lot of factors, that when we seed course version, we could seed vocab, resources, and reference guides in that same function, but that's definitely a future discussion. If we had stricter dependencies, it might not be worth it :)

> my assumption is that we'd still keep lesson-resource and lesson-vocab relationships within the .script_json

I would assume this too as these make sense to me to belong with lesson seeding.

**Review by dmcavoy — APPROVED** [review]
Sounds like everyone is in agreement that this seeding is the best set up for now and may need to be adjusted as we make improvements to course version in the future.
